File: src/commands/check.ts
import type { BeachballOptions } from '../types/BeachballOptions';
import { validateWithBump } from '../validation/validate';

export function check(options: BeachballOptions): void {
  // 'check' should include validation of dependencies, which requires the in-memory bump step
  validateWithBump(options, { checkChangeNeeded: true });
  console.log('No change files are needed');
}

File: rust/src/commands/check.rs
use anyhow::Result;

use crate::log_info;
use crate::types::options::ParsedOptions;
use crate::validation::validate::{ValidateOptions, validate};

/// Run the check command: validate that change files are present where needed.
pub fn check(parsed: &ParsedOptions) -> Result<()> {
    validate(
        parsed,
        &ValidateOptions {
            check_change_needed: true,
            check_dependencies: true,
            ..Default::default()
        },
    )?;
    log_info!("No change files are needed");
    Ok(())
}
